CMakeLists.txt

```diff
@@ -158,31 +146,34 @@ file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/plots/dalitz)
 #+++++++++++++++++++++++++++
 #        EXAMPLES          +
 #+++++++++++++++++++++++++++
-add_custom_target(examples)
-
 
+add_custom_target(examples)
 include(${Hydra_CMAKE_DIR}/AddHydraExample.cmake)
 
-add_subdirectory(examples/phase_space)
-add_subdirectory(examples/numerical_integration)
-add_subdirectory(examples/random)
-add_subdirectory(examples/fit)
-add_subdirectory(examples/histograming)
-add_subdirectory(examples/async)
-add_subdirectory(examples/misc)
-add_subdirectory(examples/phys)
```

Comment:
These should be protected with an option, makes including as part of another project easier (hint hint :) )

Reply:
In which context these files would be included in another project? I just dont see your point here. The project is header-only, there is nothing to compile and these are only example...

Reply:
You can include it inside another project, and then `add_subdirectory` on it. This would then add the CMake targets, settings, compiler flags, and functions from Hydra instead of the user being responsible for setting them him/herself. For example, that's supported with PyBind11.

PS: It is much more useful with CMake 3-style targets, as a future improvement.

Reply:
I am not sure if i understand what you mean. If it was to add a flag here, it should be set default to ON.  For include hydra in another projects, I think it is much better have something like a FindHydra.cmake...

Reply:
I did set the default to ON, unless including in another package. This is how PyBind11, fmtlib, and many other (header only) libraries work. However, you will see it done this way:

```
# Determine if fmt is built as a subproject (using add_subdirectory)
# or if it is the master project.
set(MASTER_PROJECT OFF)
if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
  set(MASTER_PROJECT ON)
endif ()
```

Would that be fine?

Reply:
Seems okay to me, but I am not familiar with this mechanism. I will check better later. Please add a comment.

Reply:
It's in the latest commit, and has a comment on what it does. :)